refactor(): remove checked property in favor of parent value (#19449)
ionic-team/ionic: BREAKING.md

The `ion-nav-push`, `ion-nav-back`, and `ion-nav-set-root` components have been removed in favor of using `ion-nav-link` with a `router-direction` property which accepts `”root”`, `“forward”`, and `“back”`. This reduces the need for maintaining multiple components when they all do the same thing with different transition directions. See the [documentation for nav-link](https://ionicframework.com/docs/api/nav-link) for more information.

#### Radio

> `ion-radio` must be used inside of an `ion-radio-group` even if there is only one `ion-radio`. Additionally, the `checked` property has been removed. Developers should set the `value` property on the respective `ion-radio-group` in order to managed checked status.

Reviewer: ```suggestion
The `ion-radio` must be used inside of an `ion-radio-group` even if there is only one `ion-radio`. Additionally, the `checked` property has been removed. Developers should set the `value` property on the parent `ion-radio-group` to match the value of the desired checked radio button.
```

> Old Usage

Reviewer: ```suggestion
Before

```

```html
<ion-radio checked>One</ion-radio>

<ion-radio-group>
  <ion-radio>One</ion-radio>
  <ion-radio checked>Two</ion-radio>
</ion-radio-group>
```

> New Usage

Reviewer: ```suggestion
After

```

<!-- TODO https://gist.github.com/brandyscarney/e6cfe43c359bb2c932e12f8d615e1669 -->

#### Segment Button

> The `checked` property has been removed. Developers should set the `value` property on the respective `ion-segment` in order to managed checked status.

Reviewer: ```suggestion
The `checked` property has been removed. Developers should set the `value` property on the parent `ion-segment` to match the value of the desired checked segment button.
```

> Old Usage

Reviewer: ```suggestion
Before

```

```html
<ion-segment>
  <ion-segment-button>One</ion-segment-button>
  <ion-segment-button checked>Two</ion-segment-button>
  <ion-segment-button>Three</ion-segment-button>
</ion-segment>
```

> New Usage

Reviewer: ```suggestion
After

```

```html
<ion-segment value="two">
  <ion-segment-button value="one">One</ion-segment-button>
  <ion-segment-button value="two">Two</ion-segment-button>
  <ion-segment-button value="three">Three</ion-segment-button>
</ion-segment>
```

> #### Select button

Reviewer: ```suggestion
#### Select Option
```

> The `selected` property has been removed. Developers should set the `value` property on the respective `ion-select` in order to managed selected status.

Reviewer: ```suggestion
The `selected` property has been removed. Developers should set the `value` property on the parent `ion-select` to match the desired selected option.
```

> Old Usage

Reviewer: ```suggestion
Before

```

```html
<ion-select>
  <ion-select-option>One</ion-select-option>
  <ion-select-option selected>Two</ion-select-option>
</ion-select>
```

> New Usage

Reviewer: ```suggestion
After

```
